client/backwards_compatibility_test.go

```diff
@@ -74,28 +75,62 @@ func Test0Dot1RepoFormat(t *testing.T) {
 
 	repo, err := NewNotaryRepository(tmpDir, gun, ts.URL, http.DefaultTransport,
 		passphrase.ConstantRetriever(passwd))
-	assert.NoError(t, err, "error creating repo: %s", err)
+	require.NoError(t, err, "error creating repo: %s", err)
+
+	// targets should have 1 target, and it should be readable offline
+	targets, err := repo.ListTargets()
+	require.NoError(t, err)
+	require.Len(t, targets, 1)
+	require.Equal(t, "LICENSE", targets[0].Name)
+
+	// delete the timestamp metadata, since the server will ignore the uploaded
+	// one and try to create a new one from scratch, which will be the wrong version
+	require.NoError(t, repo.fileStore.RemoveMeta(data.CanonicalTimestampRole))
 
 	// rotate the timestamp key, since the server doesn't have that one
 	timestampPubKey, err := getRemoteKey(ts.URL, gun, data.CanonicalTimestampRole, http.DefaultTransport)
-	assert.NoError(t, err)
-	assert.NoError(
+	require.NoError(t, err)
+	require.NoError(
 		t, repo.rootFileKeyChange(data.CanonicalTimestampRole, changelist.ActionCreate, timestampPubKey))
 
-	assert.NoError(t, repo.Publish())
+	require.NoError(t, repo.Publish())
 
-	targets, err := repo.ListTargets()
-	assert.NoError(t, err)
-	assert.Len(t, targets, 1)
-	assert.Equal(t, "v1", targets[0].Name)
+	targets, err = repo.ListTargets()
+	require.NoError(t, err)
+	require.Len(t, targets, 2)
 
 	// Also check that we can add/remove keys by rotating keys
 	oldTargetsKeys := repo.CryptoService.ListKeys(data.CanonicalTargetsRole)
-	assert.NoError(t, repo.RotateKey(data.CanonicalTargetsRole, false))
-	assert.NoError(t, repo.Publish())
+	require.NoError(t, repo.RotateKey(data.CanonicalTargetsRole, false))
+	require.NoError(t, repo.Publish())
 	newTargetsKeys := repo.CryptoService.ListKeys(data.CanonicalTargetsRole)
 
-	assert.Len(t, oldTargetsKeys, 1)
-	assert.Len(t, newTargetsKeys, 1)
-	assert.NotEqual(t, oldTargetsKeys[0], newTargetsKeys[0])
+	require.Len(t, oldTargetsKeys, 1)
+	require.Len(t, newTargetsKeys, 1)
+	require.NotEqual(t, oldTargetsKeys[0], newTargetsKeys[0])
+}
+
+// Ensures that the current client can download metadata that is published from notary 0.1 repos
+func TestDownloading0Dot1RepoFormat(t *testing.T) {
+	gun := "docker.com/notary0.1/samplerepo"
+	passwd := "randompass"
+
+	metaCache, err := store.NewFilesystemStore(
+		filepath.Join("../fixtures/compatibility/notary0.1/tuf", filepath.FromSlash(gun)),
```

Comment:
Ah, right.  :|  I will check in a new version with expiry set to like 100 years from now.